File: apps/backend/src/database/models/user-business.model.ts
import {
  Table,
  Column,
  Model,
  DataType,
  DeletedAt,
  UpdatedAt,
  CreatedAt,
  PrimaryKey,
  Default,
  BelongsTo,
  ForeignKey,
  HasMany,
  BelongsToMany,
} from 'sequelize-typescript';
import { IBusiness } from '@shared/shared/src/types/business';
import { CreationOptional, DataTypes } from 'sequelize';
import { syncStatus } from '@shared/shared/src/enums';
import { User } from './user.model';
import { Branch } from './branch.model';
import { BusinessUser } from './business-user';

@Table({ tableName: 'user_business' })
export class UserBusiness
  extends Model<IBusiness>
  implements IBusiness {
  @PrimaryKey
  @Default(DataTypes.UUIDV4)
  @Column(DataTypes.UUID)
  declare id: CreationOptional<string>;

  @Column(DataType.STRING)
  declare name: string;

  @Column(DataType.STRING)
  declare type: string;

  @Column(DataType.INTEGER)
  declare phone: string;

  @Column(DataType.STRING)
  declare street_address: string;

  @Column(DataType.STRING)
  declare city: string;

  @Column(DataType.STRING)
  declare state: string;

  @Column(DataType.STRING)
  declare country: string;

  @Column(DataType.BOOLEAN)
  declare is_verified: boolean;

  @Column(DataType.STRING)
  declare sync_status: syncStatus;

  @Column(DataType.DATE)
  declare sync_date?: string;

  @BelongsTo(() => User)
  declare user: User;

  @ForeignKey(() => User)
  @Column(DataType.UUID)
  declare user_id: string;

  @BelongsToMany(() => User, () => BusinessUser)
users: User[];

  @HasMany(() => Branch)
  declare branches: Branch

  @CreatedAt
  declare createdAt: string;

  @UpdatedAt
  declare updatedAt: string;

  @DeletedAt
  declare deletedAt?: string;
}

File: apps/spine/lib/data/services/models/user_business_model.dart

import 'package:spine/data/services/models/base_model.dart';

class UserBusiness extends BaseModel{
  final String id;
  final String userId;
  final String collectionId;
  final String name;
  final String type;
  final String phone;
  final String streetAddress;
  final String city;
  final String state;
  final String country;
  final String verification;

  final String syncDate;
  final String syncStatus;


  final String? createdAt;
  final String? updatedAt;
  final String? deletedAt;

  UserBusiness({
    required this.id,
    required this.userId,
    required this.collectionId,
    required this.name,
    required this.type,
    required this.phone,
    required this.streetAddress,
    required this.city,
    required this.state,
    required this.country,
    required this.verification,
    
    required this.syncStatus,
    required this.syncDate,


    required this.createdAt,
    required this.updatedAt,
    required this.deletedAt,
  }): super(id: id, createdAt: createdAt, deletedAt: deletedAt, syncDate: syncDate, syncStatus: syncStatus, updatedAt: updatedAt);

  factory UserBusiness.fromJson(Map<String, dynamic> json) {
    return UserBusiness(
      id: json['id'] ?? '',
      userId: json['user_id'] ?? '',
      collectionId: json['collection_id'] ?? '',
      name: json['name'] ?? '',
      type: json['type'] ?? '',
      phone: json['phone'] ?? '',
      streetAddress: json['street_address'] ?? '',
      city: json['city'] ?? '',
      state: json['state'] ?? '',
      country: json['country'] ?? '',
      verification: json['verification'] ?? '',
      
      syncStatus: json['sync_status'] ?? '',
      syncDate: json['sync_date'] ?? '',

      createdAt: json['created_at'] ?? '',
      updatedAt: json['updated_at'] ?? '',
      deletedAt: json['deleted_at'] ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'user_id': userId,
    'collection_id': collectionId,
    'name': name,
    'type': type,
    'phone': phone,
    'street_address': streetAddress,
    'city': city,
    'state': state,
    'country': country,
    'verification': verification,

    'sync_status': syncStatus,
    'sync_date': syncDate,

    'created_at': createdAt,
    'updated_at': updatedAt,
    'deleted_at': deletedAt,
  };
}
